rtil/src/threads/ue/rebo/livesplit.rs

```diff
@@ -97,4 +125,46 @@ impl Run {
     pub fn attempt_count() -> u32 {
         LIVESPLIT_STATE.lock().unwrap().timer.run().clone().attempt_count()
     }
+    pub fn save_splits(user_input: &str) -> Result<(), SplitsSaveError> {
+        let state = LIVESPLIT_STATE.lock().unwrap();
+        let valid_splits_paths = VALID_SPLITS_PATHS.lock().unwrap();
+        let path = Path::new(user_input);
+        let (path, path_display) = match (path.is_relative(), valid_splits_paths.contains(path)) {
+            (true, _) => {
+                let sanitized = sanitize_filename::sanitize(user_input);
+                (LiveSplit::splits_path().join(&sanitized), sanitized)
+            },
+            (false, true) => (path.to_owned(), user_input.to_owned()),
+            (false, false) => return Err(SplitsSaveError::DisallowedFilePath(user_input.to_string())),
+        };
+        let file = File::create(&path)
+            .map_err(|e| SplitsSaveError::CreationFailed(path_display.clone(), e.to_string()))?;
+        livesplit::save_timer(&state.timer, livesplit::IoWrite(BufWriter::new(file)))
+            .map_err(|e| SplitsSaveError::SaveFailed(path_display.clone(), e.to_string()))?;
+        Ok(())
+    }
+    pub fn load_splits(user_input: &str) -> Result<(), SplitsLoadError> {
+        let path = Path::new(user_input);
+        let sanitized = sanitize_filename::sanitize(user_input);
+        let joined_splits_path = LiveSplit::splits_path().join(sanitized);
```

Comment:
Only used within one match arm, move it into that match arm, which also allows for a better name for the latter.